src/coord/src/coord.rs

```diff
@@ -4613,6 +4634,15 @@ impl Coordinator {
         Ok(())
     }
 
+    async fn drop_secrets(&mut self, secrets: Vec<GlobalId>) {
+        let ops = secrets
+            .into_iter()
+            .map(|id| SecretOp::Delete { id })
+            .collect_vec();
+
+        self.secrets_controller.apply(ops).unwrap();
```

Comment:
this one makes me nervous. If the file gets somehow deleted on disk or if the file permissions aren't right, this will take MZ down. I don't quite want to ignore all errors here, but panicking seems excessive. I am looking for opinions.

Reply:
Why is it not safe to ignore the errors here? We have to choose between an orphaned catalog entry and an orphaned secret file. Above we chose to allow an orphaned secret file (in the create path where a crash between writing the file to disk could occur without a catalog entry). We could make the same choice here, that it is better for the file to always exist if it's in the catalog.

Otherwise, if we want to choose that it's ok for a catalog entry to exist without the file, the call to drop_secrets can move above the call to `self.catalog.transact`, where it's still safe to return an error.

Reply:
We could ignore them. Basically, any error could lead to orphaned files. 

I am mostly worried about a scenario in which the disk/FS/directory is misconfigured and these operations fail, leaving behind a couple of KB each time. That can accumulate and consume your disk without the user knowing that anything went wrong.

Reply:
I would log the error and then move on. The design doc proposed an async GC job to prevent the state from accumulating:

> The coordinator will periodically attempt to garbage collect orphaned secrets, either at startup or in a background task.